src/generator/AutoRest.Go/Templates/MethodTemplate.cshtml

```diff
@@ -193,3 +196,70 @@ func (client @(Model.Owner)) @(Model.ResponderMethodName)(resp *http.Response) (
     @:return
 @:}
 }
+
+@:@EmptyLine
+@:// @(Model.LazyMethodName) gets all results from the list and puts them inside a channel.
+@:func (client @(Model.Owner)) @(Model.LazyMethodName) @(Model.MethodParametersSignatureLazy) @(Model.MethodReturnSignatureLazy) {
+    @:respChan := make(chan @((Model.ListElement.ModelType as SequenceTypeGo).GetElement), 1)
+    @:errChan := make(chan error, 1)
+
+    @:go func() {
```

Comment:
I think in general, we can probably reduce the amount of complexity in this method. I'll chat about this with you offline.